nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/test/java/org/apache/nifi/processors/standard/TestQueryFlowFile.java

```diff
@@ -0,0 +1,392 @@
+                    Assert.assertEquals(columnNames, colNames);
+
+                    // Iterate over the rest of the records to ensure that we read the entire stream. If we don't
+                    // do this, we won't consume all of the data and as a result we will not close the stream properly
+                    Record record;
+                    while ((record = rs.next()) != null) {
+                        System.out.println(record);
+                    }
+
+                    return WriteResult.of(0, Collections.emptyMap());
+                }
+
+                @Override
+                public String getMimeType() {
+                    return "text/plain";
+                }
+
+                @Override
+                public WriteResult write(Record record, OutputStream out) throws IOException {
+                    return null;
+                }
+            };
+        }
+
+    }
+
+    private static class MockRecordWriter extends AbstractControllerService implements RecordSetWriterFactory {
+        private final String header;
+
+        public MockRecordWriter(final String header) {
+            this.header = header;
+        }
+
+        @Override
+        public RecordSetWriter createWriter(final ComponentLog logger) {
+            return new RecordSetWriter() {
+                @Override
+                public WriteResult write(final RecordSet rs, final OutputStream out) throws IOException {
+                    out.write(header.getBytes());
+                    out.write("\n".getBytes());
+
+                    int recordCount = 0;
+                    final int numCols = rs.getSchema().getFieldCount();
+                    Record record = null;
+                    while ((record = rs.next()) != null) {
+                        recordCount++;
+                        int i = 0;
+                        for (final String fieldName : record.getSchema().getFieldNames()) {
+                            final String val = record.getAsString(fieldName);
+                            out.write("\"".getBytes());
+                            out.write(val.getBytes());
+                            out.write("\"".getBytes());
+
+                            if (i++ < numCols - 1) {
+                                out.write(",".getBytes());
+                            }
+                        }
+                        out.write("\n".getBytes());
+                    }
+
+                    return WriteResult.of(recordCount, Collections.emptyMap());
+                }
+
+                @Override
+                public String getMimeType() {
+                    return "text/plain";
+                }
+
+                @Override
+                public WriteResult write(Record record, OutputStream out) throws IOException {
+                    return null;
+                }
+            };
+        }
+    }
+
+    private static class MockRecordParser extends AbstractControllerService implements RowRecordReaderFactory {
```

Comment:
Can this be its own class for reuse across the processors that will use a RowRecordReaderFactory?